### PR TITLE
Add golangci lint

### LINKED ISSUE #34: Enable golangci-lint in repo
We should:
* add a golangci-lint configuration
* update the makefile to trigger golangci-lint
* Enable the golangci-lint checks in PRs

### PR DESCRIPTION
**Pull request by elezar**
This adds golangci-lint and addresses the issues found.

Closes #34

### CODE REVIEW

**Comment by elezar** [comment]
This includes the change from #32.

File: Makefile [hunk 1]
```diff
@@ -66,43 +66,24 @@ fmt:
...
 COVERAGE_FILE := coverage.out
 test: build cmds
-	go test -v -coverprofile=$(COVERAGE_FILE) $(MODULE)/...
+	go test -coverprofile=$(COVERAGE_FILE) $(MODULE)/cmd/... $(MODULE)/internal/... $(MODULE)/api/...
 
 coverage: test
 	cat $(COVERAGE_FILE) | grep -v "_mock.go" > $(COVERAGE_FILE).no-mocks
 	go tool cover -func=$(COVERAGE_FILE).no-mocks
 
+generate:
```
Comment (ArangoGutierrez) [inline Makefile:84]: ...?  is this needed for the PR ref? (add golangci-lint)

Comment (elezar) [reply Makefile:84]: No, but id does align the makefile with other projects.

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**